src/crypto/crypto_keys.cc

```diff
@@ -1134,6 +1136,46 @@ void KeyObjectHandle::InitEDRaw(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(true);
 }
 
+void KeyObjectHandle::Equals(const FunctionCallbackInfo<Value>& args) {
+  std::shared_ptr<KeyObjectData> key =
+    Unwrap<KeyObjectHandle>(args.Holder())->Data();
+  std::shared_ptr<KeyObjectData> key2 =
+    Unwrap<KeyObjectHandle>(args[0].As<Object>())->Data();
+
+  KeyType keyType = key->GetKeyType();
+  CHECK_EQ(keyType, key2->GetKeyType());
+
+  bool ret;
+  switch (keyType) {
+    case kKeyTypeSecret: {
+      size_t size = key->GetSymmetricKeySize();
+      if (size == key2->GetSymmetricKeySize())
+        ret = CRYPTO_memcmp(
+          key->GetSymmetricKey(),
+          key2->GetSymmetricKey(),
+          size) == 0;
+      else
+        ret = false;
+      break;
+    }
+    case kKeyTypePublic:
+      // Fall through
+    case kKeyTypePrivate: {
+      EVP_PKEY* fkey = key->GetAsymmetricKey().get();
+      EVP_PKEY* fkey2 = key2->GetAsymmetricKey().get();
+      if (EVP_PKEY_id(fkey) == EVP_PKEY_id(fkey2))
+        ret = EVP_PKEY_eq(fkey, fkey2) == 1;
+      else
+        ret = false;
```

Comment:
```suggestion
      if (EVP_PKEY_id(pkey) == EVP_PKEY_id(pkey2)) {
        #if OPENSSL_VERSION_MAJOR >= 3
          ret = EVP_PKEY_eq(pkey, pkey2) == 1;
        #else
          ret = EVP_PKEY_cmp(pkey, pkey2) == 1;
        #endif
      } else {
        ret = false;
      }
```

@tniessen Please see the [`EVP_PKEY_eq`(3) / `EVP_PKEY_cmp`(1.1.1)](https://www.openssl.org/docs/manmaster/man3/EVP_PKEY_cmp.html) notes.

I'm not sure what to do when running with linked 1.1.1 (which full CI will fail with). Would like your take on it. I can use `EVP_PKEY_cmp` instead (_cmp is an alias for _eq in 3.x) and we already make sure that the key types are equal before comparing.